legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java

```diff
@@ -163,40 +264,26 @@ private static MappingModelCoverageAnalysisResult buildMappingModelCoverageAnaly
+                }).filter(Objects::nonNull).collect(Collectors.toList()));
                 Root_meta_analytics_binding_modelCoverage_BindingModelCoverageAnalysisResult bindingAnalysisResult = core_analytics_binding_modelCoverage_analytics.Root_meta_analytics_binding_modelCoverage_getBindingModelCoverage_Binding_MANY__BindingModelCoverageAnalysisResult_1_(bindings, pureModel.getExecutionSupport());
-                List<String> functionPaths = pureModelContextData.getElements().stream().filter(el -> el instanceof Function).map(e -> e.getPath()).collect(Collectors.toList());
-                List<String> allExtraElements = functionPaths;
-                allExtraElements.add(dataSpaceProtocol.getPath());
-                pureModelContextData.getElements().stream().filter(el -> allExtraElements.contains(el.getPath())).forEach(builder::addElement);
-                List<org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement> elements = builder.build().getElements();
-                RichIterable<? extends ConcreteFunctionDefinition<? extends  Object>> functions = org.eclipse.collections.api.factory.Lists.mutable.ofAll(functionPaths.stream().map(path ->
+                // only keep DataSpaceTemplateExecutable
+                if (dataSpaceProtocol.executables != null)
                 {
-                    ConcreteFunctionDefinition<? extends Object> function = null;
-                    try
-                    {
-                        function = pureModel.getConcreteFunctionDefinition_safe(path);
-                        if (function == null)
-                        {
-                            Function _function = (Function) elements.stream().filter(e -> e.getPath().equals(path)).findFirst().get();
-                            function = pureModel.getConcreteFunctionDefinition_safe(path + HelperValueSpecificationGrammarComposer.getFunctionSignature(_function));
-
-                        }
-                        return function;
-                    }
-                    catch (Exception ignored)
-                    {
-
-                    }
-                    return null;
-                }).filter(c -> c != null).collect(Collectors.toList()));
-                Root_meta_analytics_function_modelCoverage_FunctionModelCoverageAnalysisResult functionCoverageAnalysisResult = core_analytics_function_modelCoverage_analytics.Root_meta_analytics_function_modelCoverage_getFunctionModelCoverage_ConcreteFunctionDefinition_MANY__FunctionModelCoverageAnalysisResult_1_(org.eclipse.collections.impl.factory.Lists.mutable.ofAll(functions), pureModel.getExecutionSupport());
+                    DataSpace updatedDataSpaceProtocol = new DataSpace(dataSpaceProtocol);
+                    updatedDataSpaceProtocol.executables = updatedDataSpaceProtocol.executables.stream().filter(ex -> ex instanceof DataSpaceTemplateExecutable).collect(Collectors.toList());
```

Comment:
Dont think we should add dataspace to this mini graph. we should keep to pmcd of mapped elements and not add anything else. The root class should have all the information we need.